test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -487,6 +491,361 @@ public async Task WhenInjectingStreamBasedJsonWriterFactoryAsync_CreatesWriterUs
         #endregion "ODataUtf8JsonWriter support"
 #endif
 
+#if NETCOREAPP3_1_OR_GREATER
+        #region "ODataJsonElementValue support"
+        [Fact]
+        public void WriteEntityWithJsonElementValues()
+        {
+            // Arrange
+            EdmModel model = new EdmModel();
+            EdmEntityType personType = model.AddEntityType("ns", "Person", baseType: null, isAbstract: false, isOpen: true);
+            personType.AddKeys(
+                personType.AddStructuralProperty("Id", EdmPrimitiveTypeKind.Int32));
+            personType.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String);
+            personType.AddStructuralProperty(
+                "Emails",
+                new EdmCollectionTypeReference(
+                    new EdmCollectionType(EdmCoreModel.Instance.GetString(isNullable: false))));
+            EdmComplexType addressType = model.AddComplexType("ns", "Address");
+            addressType.AddStructuralProperty("City", EdmPrimitiveTypeKind.String);
+            addressType.AddStructuralProperty("Country", EdmPrimitiveTypeKind.String);
+            personType.AddStructuralProperty("Address", new EdmComplexTypeReference(addressType, isNullable: false));
+            IEdmEntitySet peopleSet = model.AddEntityContainer("ns", "Service").AddEntitySet("People", personType);
+
+
+            string source = "{" +
+                @"""Id"": 1," +
+                @"""Name"": ""John""," +
+                @"""Emails"":[""john@mailer.com"",""john@work.com""]," +
+                @"""Address"":{""City"":""Nairobi"",""Country"":""Kenya""}" +
+                @"}";
+            var json = JsonDocument.Parse(source);
+            var jsonRoot = json.RootElement;
+            var resource = new ODataResource
+            {
+                Properties = new[]
+                {
+                    new ODataProperty { Name = "Id", Value = new ODataJsonElementValue(jsonRoot.GetProperty("Id")) },

```

Comment:
why don't we use the "int" directly?
I mean what's the benefit to use "ODataJsonElementValue" to wrap the "Id", "Name"... (such primitive type property)?

Reply:
Frankly, there's no value in wrapping the primitive here with `ODataJsonElementValue`, but this is how I was able to reproduce the bug reported by AGS. I can check back with AGS whether they have a more representative repro.